src/RouteCollectorModifier.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace FastRoute;
+
+class RouteCollectorModifier extends \FastRoute\RouteCollector
+{
+    /** @var array */
+    private $currentModifiers = [];
```

Comment:
Why do you need `$currentModifiers` if this array stays empty?

Line 45 `$this->currentModifiers = $previousModifiers;` makes sure that `$currentModifiers` will stay empty.

The `foreach` in `::addRoute()` will also do nothing.

You should add unit tests and verify that this is doing what you think it is doing.

Reply:
Take a look at the how the `addGroup` method works in [RouteCollector.php](https://github.com/nikic/FastRoute/blob/master/src/RouteCollector.php#L57). This is doing a very similar thing.

This does do what I think it does, but you're correct, tests do need to be written, however as I mentioned in my original post:
> This is not ready to be integrated, changes would need to be made to integrate this properly, but before I spend more time on this, I'd like some opinions :)